testutils/src/main/scala/za/co/absa/ComparisonJob.scala

```diff
@@ -0,0 +1,65 @@
+package za.co.absa
+
+import org.apache.spark.sql._
+import org.apache.spark.SparkContext
+import org.apache.spark.sql.types.{StructField, StructType}
+import za.co.absa.enceladus.standardization.CmdConfig
+
+object ComparisonJob {
+  def getReader(cmd: CmdConfig)(implicit sparkSession: SparkSession): DataFrameReader = {
+    val dfReader = sparkSession.read.format(cmd.rawFormat)
+    // applying format specific options
+    val  dfReader4= {
+      val dfReader1 = if (cmd.rowTag.isDefined) dfReader.option("rowTag", cmd.rowTag.get) else dfReader
+      val dfReader2 = if (cmd.csvDelimiter.isDefined) dfReader1.option("delimiter", cmd.csvDelimiter.get) else dfReader1
+      val dfReader3 = if (cmd.csvHeader.isDefined) dfReader2.option("header", cmd.csvHeader.get) else dfReader2
+      dfReader3
+    }
+    if (cmd.rawFormat.equalsIgnoreCase("fixed-width")) {
```

Comment:
We can probably refactor this a little to make it a little more organized in the same block as the dfReaders above.

Reply:
Yeah was thinking the same :-)

Reply:
New commit added and with it a new class that solves this